packages/client/src/features/smart-connector/smart-connector.ts

```diff
@@ -366,11 +381,25 @@ export class SmartConnector extends AbstractUIExtension implements IActionHandle
         this.navigateSearchField(event, itemGroup);
     }
 
+    protected handlerHeaderKey(event: KeyboardEvent, groupItems: HTMLElement, header: HTMLElement) {
+        this.navigateHeader(event, groupItems, header);
+        this.closeOnEscapeKey(event);
+    }
+
+    protected closeOnEscapeKey(event: KeyboardEvent) {
+        if (matchesKeystroke(event, 'Escape')) {
+            this.hideSmartConnector();
+            // assumes that the graph is always the last child of base div
+            // this focus is done to "reactivate" the key listener to re-open if needed
+            document.getElementById(this.options.baseDiv)?.last().focus();
```

Comment:
Since the focus is lost when the connector is closed, the KeyListener does not react to any inputs. It does also not react when any tool is selected. Because of this, I just added a onkeydown event on every tool and when it closes, it refocuses on the base div (the graph). Is this assumption here correct? Or is there a better way to get the base div?

Reply:
Because the KeyListener does not react anymore. So if I, for example, open the connector with space, close it again with escape and then try to open it again, it won't work